Sources/TuistSupport/Commands/Command.swift

```diff
@@ -19,6 +19,20 @@ public protocol Command {
     /// - Parameter arguments: Result of parsing the arguments that the user passed to the CLI.
     /// - Throws: Errors that are thrown by the underlying command action.
     func run(with arguments: ArgumentParser.Result) throws
+
+    /// If needed, command can do custom parsing
+    /// As default, result from `parser.parse` is used
+    /// - Parameters:
+    ///     - parser: Default parser
+    ///     - argumets: List of arguments that the user passed to the CLI
+    /// - Returns: Result of parsing the arguments that the user passed to the CLI.
+    func parse(with parser: ArgumentParser, arguments: [String]) throws -> ArgumentParser.Result
```

Comment:
Does it need to be part of the protocol?

Reply:
Yes, otherwise I'd need to rewrite [this](https://github.com/tuist/tuist/pull/1092/files/fef4af1f612a7045b1869faecf57e9b64d5c0278#diff-7e3710f4470c93590881d6a91fdd047eR112)